core/src/main/java/com/linecorp/armeria/server/AnnotatedHttpServiceMethod.java

```diff
@@ -214,9 +215,14 @@ private HttpResponse convertResponse(ServiceRequestContext ctx, @Nullable Object
                 }
             }
         }
+        // There is no response converter which is able to convert 'null' result to a response.
+        // In this case, we deal the result as '204 No Content' instead of '500 Internal Server Error'.
+        if (result == null) {
+            return HttpResponse.of(HttpStatus.NO_CONTENT);
+        }
         throw new IllegalStateException(
                 "No response converter exists for a result: " +
-                result != null ? result.getClass().getSimpleName() : "(null)");
+                (result != null ? result.getClass().getSimpleName() : "(null)"));
```

Comment:
Then we can simply do `"No response converter exists for a result: " + result.getClass().getSimpleName()`

Reply:
Oops! You're right. Thanks!